lib/chef/provider/cron.rb

```diff
@@ -89,11 +89,18 @@ def load_current_resource
       end
 
       def cron_different?
+        property_with_environment?
         CRON_ATTRIBUTES.any? do |cron_var|
           new_resource.send(cron_var) != current_resource.send(cron_var)
         end
       end
 
+      # Raises warning if user is using a `environment` for an entry that can also be specified as a `property`
+      def property_with_environment?
```

Comment:
@vijaymmali1990 we need to revisit this this is wrong implementation of method in ruby method with ? should always return boolean type.